salt/orch/etcd-migrate.sls

```diff
@@ -0,0 +1,107 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_grains:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+       - salt: update_grains
+
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master', fun='network.ip_addrs', tgt_type='compound') %}
+{%- for master_id in masters.keys() %}
+
+{{ master_id }}-stop-services:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kube-apiserver.stop
+      - kube-controller-manager.stop
+      - kube-scheduler.stop
+      - etcd.stop
+
+{% endfor %}
+
+{%- set workers = salt.saltutil.runner('mine.get', tgt='G@roles:kube-minion', fun='network.ip_addrs', tgt_type='compound') %}
+{%- for worker_id, ip in workers.items() %}
+
+{{ worker_id }}-stop-services:
+  salt.state:
+    - tgt: {{ worker_id }}
+    - sls:
+      - kubelet.stop
+      - kube-proxy.stop
+      - etcd.stop
+
+{% endfor %}
+
+{%- for master_id in masters.keys() %}
```

Comment:
Again, I'm still not sure I understand why were doing this in a loop like this? These steps look to be totally independent on each node, we should drop the masters+minions loops, and just target all masters and minions in 1 go.

Reply:
(with a batch size of something reasonable, like 15 nodes)

Reply:
Yes, as I said before there is reason for this. Master and Worker nodes they have different sets of services. And first we would like to stop kube-apiservice first

Reply:
The set of commands inside both of these loops are identical for masters and workers, and we're doing masters - 1 at a time and workers - 1 at a time. There is no need to do these one at a time, and I believe no need to do masters first, then workers second - as the commands done within the loops are entirely independent - all this does is extend the duration of the downtime.